projects/packages/my-jetpack/src/js-static/jp-js-error-tracker/index.js

```diff
@@ -0,0 +1,323 @@
+/**
+ * JPJSErrorTracker - Simplified JavaScript error tracking library
+ * Captures all JavaScript errors and passes them to a callback
+ * @param {object} global - The global object (window in browsers)
+ */
+( function ( global ) {
+	/**
+	 * JPJSErrorTracker constructor
+	 * @param {Function} onError - Callback function that receives error data
+	 */
+	function JPJSErrorTracker( onError ) {
+		if ( typeof onError !== 'function' ) {
+			throw new Error( 'JPJSErrorTracker requires an onError callback function' );
+		}
+
+		this.onError = onError;
+		this.sessionId = this.generateSessionId();
+		this.pageLoadTime = Date.now();
+		this.errorCount = 0;
```

Comment:
I wonder if we should have a `maxErrors` as well, rolling off the oldest errors to prevent this from getting too large. However, in order to do that properly we'd need an accurate stack of all errors, not just unique ones.